src/Microsoft.VisualStudio.Mac.LanguageServices.Razor/VisualStudioErrorReporter.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Razor;
+using MonoDevelop.Core;
+
+namespace Microsoft.VisualStudio.Mac.LanguageServices.Razor
+{
+    internal class VisualStudioErrorReporter : ErrorReporter
+    {
+        public override void ReportError(Exception exception)
+        {
+            if (exception == null)
```

Comment:
Should this really be ignored ? Is this the functionality of the base class?
My point is that a method shouldn't even be called, if it doesn't make sense. So, in that case, throwing would make more sense.
Again, I understand that it may be a breaking change if implemented that way.

Reply:
In our cases yes. This is a non-critical public API that's available to any extension in VisualStudio (that depends on Razor) and throwing from here results in the IDE crashing. In the tooling side of Razor we default to being more protective in non-critical APIs so we don't tarnish the user experience.

In any case. I will definitely add a `Debug.Assert` here so in a debugging VS scenario we get proper notifications that something unexpected is happening. 👍